tests/unit/routes/shield-round4-enhancements.test.js

```
@@ -0,0 +1,630 @@
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Enhanced Input Validation', () => {
+    test('should handle null/undefined query parameters safely', async () => {
+      supabaseServiceClient.from.mockReturnValue({
+        select: jest.fn().mockReturnThis(),
+        eq: jest.fn().mockReturnThis(),
+        order: jest.fn().mockReturnThis(),
+        range: jest.fn().mockReturnThis(),
+        then: jest.fn().mockResolvedValue({
```

Biome (2.1.2)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)